feat(linter): support allowable method diagnostic for `eslint/no-console`

ityuany opened the pull request:
- Modified `no_console_diagnostic` function to add a list of allowed methods argument
- Include the list of allowed console methods in the prompt message.
- Updated diagnostic calls to support passing information about allowed methods.

ityuany committed:
feat(eslint): improve the no-console rule to support allowable method configuration

- Modified `no_console_diagnostic` function to add a list of allowed methods argument
- Include the list of allowed console methods in the prompt message.
- Updated diagnostic calls to support passing information about allowed methods.

shulaoda commented on crates/oxc_linter/src/rules/eslint/no_console.rs:
```diff
@@ -10,10 +10,13 @@ use crate::{
-        .with_help("Delete this console statement.")
+fn no_console_diagnostic(span: Span, allow: &Vec<CompactStr>) -> OxcDiagnostic {
+    OxcDiagnostic::warn(format!(
+        "eslint(no-console): Unexpected console statement. Only supported methods: {}",
+        allow.iter().map(|s| s.to_string()).collect::<Vec<String>>().join(", ")
```
When `allow` is empty, `Only supported methods` should not be prompted.

ityuany replied:
yes , You're right.
fixed.
thanks.

ityuany committed:
refactor(linter): improve warning message construction in no_console rules

Refactored warning message construction logic in no_console rules to optimize message formatting when the allow list is empty.

ityuany committed:
refactor(eslint): optimizes string handling in no_console rules

- Replacing `Vec` references with slices to improve performance
- Replace `to_string` with `String::new` to optimize empty string creation.
- Apply formatted strings to improve readability

ityuany committed:
fix(linter): update eslint_no_console.snap to specify supported console methods

camc314 commented on crates/oxc_linter/src/rules/eslint/no_console.rs:
```diff
@@ -10,8 +10,17 @@ use crate::{
+    } else {
+        format!(
+            "Only supported methods: {}",
+            allow.iter().map(ToString::to_string).collect::<Vec<String>>().join(", ")
```
```suggestion
            allow.iter().map(oxc_span::CompactStr::as_str).join(",")
```
we can make this a bit better - we don't need to `collect` or convert to `Strings` before joining

ityuany committed:
fix(no_console): Update the formatting string to remove unnecessary string conversions.

camc314 commented on crates/oxc_linter/src/rules/eslint/no_console.rs:
```diff
@@ -10,8 +10,17 @@ use crate::{
+    } else {
+        format!(
+            "Only supported methods: {}",
+            allow.iter().map(oxc_span::CompactStr::as_str).join(",")
```
oops `format!("Only supported methods: {}", allow.join(", "))` looks better

ityuany replied:
You are right.

ityuany committed:
fix(linter): Fix the string concatenation issue in the no_console rule

- Modify the string concatenation logic in the `no_console.rs` file to uniformly use the `join(",")` method.

- Update the corresponding snapshot tests to reflect these changes, ensuring no extra spaces are included.
